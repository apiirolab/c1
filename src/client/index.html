<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>