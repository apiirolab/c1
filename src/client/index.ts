export const info = 'Client Info';
