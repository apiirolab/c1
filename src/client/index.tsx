import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
