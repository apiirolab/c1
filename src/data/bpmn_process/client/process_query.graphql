query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
