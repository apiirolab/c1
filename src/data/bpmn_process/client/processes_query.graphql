query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    id
    name
  }
}
