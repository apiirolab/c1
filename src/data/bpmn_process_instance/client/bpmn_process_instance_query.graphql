fragment BpmnProcessResource on Resource {
  name
  type
  id
  content
}

fragment BpmnProcessComment on Comment {
  id
  date
  replyTo
  text
  user {
    name
  }
}

fragment BpmnProcessTask on BpmnTaskInstance {
  name
  dateStarted
  dateFinished
  performer {
    name
  }
  performerRoles
}

fragment BpmnProcessDefinition on BpmnProcess {
  name
  description
  type
  model
  resources {
    ...BpmnProcessResource
  }
}

query BpmnProcessInstanceQuery($id: String!) {
  bpmnProcessInstance(id: $id) {
    id
    status
    process {
      ...BpmnProcessDefinition
    }
    owner {
      name
    }
    tasks {
      ...BpmnProcessTask
    }
    comments {
      ...BpmnProcessComment
    }
    resources
    dateStarted
    dateFinished
  }
}
