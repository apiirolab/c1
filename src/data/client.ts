import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
