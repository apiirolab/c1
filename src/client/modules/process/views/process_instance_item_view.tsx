import * as React from 'react';

import { Link } from '@reach/router';
import { List } from 'semantic-ui-react';

import { QueryTypes } from 'data/client';
import styled from 'styled-components';
import { ProcessIcon } from './process_icon';
import { ProcessInstanceStatus } from './process_instance_status';

type Props = {
  processInstance: QueryTypes.BpmnProcessInstancesQuery_BpmnProcessInstances;
  context: App.Context;
};

const Meta = styled.div`
  color: #999;
  margin: 6px 0px;
`;

const Info = styled.div`
  margin: 3px 0px;
`;

export const ProcessInstanceItem: React.SFC<Props> = ({ processInstance, context }) => {
  const waitingTasks = processInstance.tasks.filter(t => !t.dateFinished);
  const waitingText = waitingTasks
    .map(
      t =>
        context.i18n`${t.performerRoles.join(' or ')} to ${t.name} since ${context.Ui.relativeDate(
          t.dateStarted
        )}`
    )
    .join(' and ');
  const completedTasks = processInstance.tasks.filter(t => !!t.dateFinished);
  const lastTask = completedTasks.length > 0 ? completedTasks.sort()[0] : null;

  return (
    <List.Item>
      <ProcessIcon type={processInstance.process.type} />
      <List.Content verticalAlign="top">
        <List.Header as={Link} to="nowhere">
          {processInstance.process.name}
        </List.Header>
        <List.Description>
          <Meta>
            <ProcessInstanceStatus status={processInstance.status} context={context} />{' '}
            &nbsp;&middot;&nbsp;
            {context.i18n`Started ${context.Ui.relativeDate(processInstance.dateStarted)} by ${
              processInstance.owner.name
            }`}
            <If condition={processInstance.dateFinished}>
              <>
                &nbsp;&middot;&nbsp;
                {context.i18n`Finished ${context.Ui.relativeDate(processInstance.dateFinished)}`}
              </>
            </If>
          </Meta>

          <If condition={!!lastTask}>
            <Info>
              <b>{context.i18n`Last activity: `}</b>
              {context.i18n`${lastTask.performer.name} performed "${
                lastTask.name
              }" ${context.Ui.relativeDate(lastTask.dateFinished)}`}
            </Info>
          </If>

          <If
            condition={
              processInstance.status === QueryTypes.BpmnProcessInstanceStatus.Running &&
              !!waitingText
            }
          >
            <Info>
              <b>{context.i18n`Waiting for: `}</b> {waitingText}
            </Info>
          </If>
        </List.Description>
      </List.Content>
    </List.Item>
  );
};
