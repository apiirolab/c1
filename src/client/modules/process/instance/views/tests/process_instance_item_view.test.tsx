import * as React from 'react';

import { create, render } from 'client/tests';

import { createProcessListInstance } from 'client/modules/process/common/tests/process_test_data';
import { List } from 'semantic-ui-react';
import { ProcessInstanceItem } from '../process_instance_item_view';

describe('Process Instance', () => {
  describe('Item', () => {
    const context = create.context();

    it('renders correctly', () => {
      const processInstance = createProcessListInstance();
      const component = render(
        <ProcessInstanceItem processInstance={processInstance} context={context} />
      );
      expect(component).toMatchSnapshot();
    });

    return {
      component: (
        <List>
          <ProcessInstanceItem processInstance={createProcessListInstance()} context={context} />
        </List>
      )
    };
  });
});
