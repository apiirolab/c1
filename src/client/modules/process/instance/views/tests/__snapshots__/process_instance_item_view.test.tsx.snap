// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Instance Item renders correctly 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <i
    aria-hidden="true"
    className="money icon middle aligned sc-bdVaJa jnSeCf"
  />
  <div
    className="top aligned content"
  >
    <a
      className="header"
      href="/process/process-1/view/information/1"
    >
      Process 1
    </a>
    <div
      className="description"
    >
      <div
        className="sc-bwzfXH gzggMh"
      >
        <div
          className="ui green tiny label"
          onClick={[Function]}
        >
          Running
        </div>
         
         · 
        Started 2 months ago by Buddy Holly
      </div>
      <div
        className="sc-htpNat eaijOI"
      >
        <b>
          Last activity: 
        </b>
        Tomas Trescak performed "Task Name" 2 months ago
      </div>
      <div
        className="sc-htpNat eaijOI"
      >
        <b>
          Waiting for: 
        </b>
         
        Purchasing to Complete Purchase since 2 months ago and IT Support or IT Delivery to Deliver Technology since 2 months ago
      </div>
    </div>
  </div>
</div>
`;
